src/client/mod.rs

```diff
@@ -320,26 +336,75 @@ impl<'a, T> VaultClient<'a, T>
     }
 
     /// Revoke `VaultClient`'s token. This token can no longer be used.
+    /// Corresponds to [`/auth/token/revoke-self`][token].
+    ///
+    /// ```no_run
+    /// # extern crate hashicorp_vault as vault;
+    /// # use vault::Client;
+    /// # fn main() {
+    /// let host = "http://127.0.0.1:8200";
+    /// let token = "test12345";
+    /// let mut client = Client::new(host, token).unwrap();
+    ///
+    /// client.revoke().unwrap();
+    /// # }
+    /// ```
+    ///
+    /// [token]: https://www.vaultproject.io/docs/auth/token.html
     pub fn revoke(&mut self) -> Result<()> {
-        let _ = try!(self.post(&format!("{}/v1/auth/token/revoke-self", self.host), None));
+        let _ = try!(self.post("/v1/auth/token/revoke-self", None));
         Ok(())
     }
 
-    /// Renew a specific lease that your token controls
-    /// https://www.vaultproject.io/docs/http/sys-renew.html
+    /// Renew a specific lease that your token controls.  Corresponds to
+    /// [`/v1/sys/renew`][renew].
+    ///
+    /// ```no_run
+    /// # extern crate hashicorp_vault as vault;
+    /// # use vault::Client;
+    /// # fn main() {
+    /// let host = "http://127.0.0.1:8200";
+    /// let token = "test12345";
+    /// let mut client = Client::new(host, token).unwrap();
+    ///
+    /// // TODO: Right now, we offer no way to get lease information for a
```

Comment:
Yes. I've filed a separate issue in #7, which will also allow us to make this doc test runnable.